alf/algorithms/sac_algorithm.py

```diff
@@ -784,10 +784,19 @@ def _critic_train_step(self, inputs: TimeStep, state: SacCriticState,
         return state, info
 
     def _alpha_train_step(self, log_pi):
-        alpha_loss = nest.map_structure(
-            lambda la, lp, t: la * (-lp - t).detach(), self._log_alpha, log_pi,
-            self._target_entropy)
-        return sum(nest.flatten(alpha_loss))
+        if self._act_type == ActionType.Mixed:
+            alpha_loss = 0
+            # in the mixed action case, both log_pi and _target_entropy are tuples of two
+            # elements, corresponding to the discrete action and continuous action respectively
+            for i in range(2):
+                neg_entropy = sum(nest.flatten(log_pi[i]))
+                alpha_loss = self._log_alpha * (
```

Comment:
+=

Reply:
Sorry the previous fix was not done correctly. Fixed the loss aggregation and also added the missing index to ``self._log_alpha``.